### PR TITLE
Issue 373: Fix class path scanner subclass check

### PR DESCRIPTION
**Pull request by alanfx**
Use correct method to find subclasses and interface implementors

### CODE REVIEW

File: core/src/main/java/org/radargun/config/ClasspathScanner.java [hunk 1]
```diff
@@ -25,29 +26,40 @@ private ClasspathScanner() {
...
-            if (superClass != null && !superClass.isAssignableFrom(clazz)) {
-               continue;
-            }
             consumer.accept((Class<? extends TClass>) clazz);
          } catch (Throwable e) {
             // static ctor can throw non-wrapped error
-            log.error("Cannot load class " + className, e);
-            continue;
+            log.trace("Cannot load class " + className, e);
+            return;
```
Comment (rvansa) [inline core/src/main/java/org/radargun/config/ClasspathScanner.java]: Why do you want to quit the lookup once single class throws?

**Comment by rvansa** [comment]
Hmm, I didn't realize that I could just compute the intersection of those two sets. Good, go for it.

File: core/src/main/java/org/radargun/config/ClasspathScanner.java (hunk 1 above)
Comment (alanfx) [inline core/src/main/java/org/radargun/config/ClasspathScanner.java]: I don't! Just keep me away from the code...

**Comment by alanfx** [comment]
Hang on, I realized why the name of the superclass needs to be in the constructor

**Comment by alanfx** [comment]
Well, I guess I didn't. I added the check for `assignableFrom.size() > 0`, because `getNamesOfSubclassesOf` returns 0 classes when the superclass is `java.lang.Object`. I thought that whitelisting that class in the constructor would change that, but it didn't help, so I didn't make any other changes.

**Comment by alanfx** [comment]
Ok, this latest version skips checking for the superclass if `null` is passed to `ClasspathScanner.scanClasspath`

**Pull request merged**